src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,295 @@
+     */
+    public static Map<ZonedDateTime, String> getItineraryExistenceQueries(MonitoredTrip trip, boolean checkAllDays)
+        throws URISyntaxException {
+        return getQueriesFromDates(
+            excludeRealtime(trip.parseQueryParams()),
+            getDatesToCheckItineraryExistence(trip, checkAllDays)
+        );
+    }
+
+    /**
+     * @return a copy of the specified query parameter map, with ignoreRealtimeUpdates set to true.
+     */
+    public static Map<String, String> excludeRealtime(Map<String, String> params) {
+        Map<String, String> result = new HashMap<>(params);
+        result.put(IGNORE_REALTIME_UPDATES_PARAM, "true");
+        return result;
+    }
+
+    /**
+     * Checks that, for each query provided, an itinerary exists.
+     * @return An object with a map of results and summary of itinerary existence.
+     */
+    public static ItineraryExistence checkItineraryExistence(MonitoredTrip trip, boolean checkAllDays, boolean sortDates) throws URISyntaxException {
+        // Get queries to execute by date.
+        Map<ZonedDateTime, String> queriesByDate = ItineraryUtils.getItineraryExistenceQueries(trip, checkAllDays);
+        // TODO: Consider multi-threading?
+        Map<ZonedDateTime, Itinerary> datesWithMatchingItineraries = new HashMap<>();
+
+        Collection<Map.Entry<ZonedDateTime, String>> entriesToIterate = sortDates
+            ? queriesByDate.entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(Collectors.toList())
+            : queriesByDate.entrySet();
+
+        for (Map.Entry<ZonedDateTime, String> entry : entriesToIterate) {
+            ZonedDateTime date = entry.getKey();
+            String params = entry.getValue();
+
+            OtpDispatcherResponse response = OtpDispatcher.sendOtpPlanRequest(params);
+            TripPlan plan = response.getResponse().plan;
+            if (plan != null && plan.itineraries != null) {
+                for (Itinerary itinerary: plan.itineraries) {
+                    // If a matching itinerary is found, save the date with the matching itinerary.
+                    // The matching itinerary will replace the original trip.itinerary.
+                    if (itinerariesMatch(trip.itinerary, itinerary)) {
+                        datesWithMatchingItineraries.put(date, itinerary);
+                    }
+                }
+            }
+        }
+        return new ItineraryExistence(queriesByDate.keySet(), datesWithMatchingItineraries);
+    }
+
+    /**
+     * Overload used in normal conditions.
+     */
+    public static ItineraryExistence checkItineraryExistence(MonitoredTrip trip, boolean checkAllDays) throws URISyntaxException {
+        return checkItineraryExistence(trip, checkAllDays,false);
+    }
+
+    /**
+     * TODO: fully implement
+     */
+    public static boolean itineraryIsSavable(Itinerary itinerary) {
```

Comment:
FYI @evansiroky, in #91 I called this method `ItineraryCanBeMonitored`.

Reply:
Should we make this PR dependent on #91 then?

Reply:
@evansiroky leave this function as is for now, although there will be a merge conflict... As of https://github.com/ibi-group/otp-middleware/commit/83405f10b950ed5b4fb6ca75ddfa1a1f34753d1e, the equivalent of `itineraryIsSaveable` doesn't exist anymore (I broke it down per your comments).